### ISSUE TITLE
[Improvement]: Upgrade to Doctrine DBAL v3

### ISSUE THREAD

**Issue by brusch** [pimcore/customer-data-framework#325]
### Improvement description

Ideally support both v2 and v3.

**Comment by brusch** [comment]
This is currently breaking the Pimcore 11 demo: 

```
  - Installing pimcore/demo (11.x-dev 65c4690): Extracting archive
Created project in /home/demos/build/opensource-master/source-master
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires pimcore/customer-management-framework-bundle ^3.3.0 -> satisfiable by pimcore/customer-management-framework-bundle[v3.3.0, v3.3.1, 3.3.x-dev, 3.x-dev].
    - pimcore/customer-management-framework-bundle[v3.3.0, ..., 3.x-dev] require doctrine/dbal ^2.13.1 -> satisfiable by doctrine/dbal[2.13.1, ..., 2.13.9].
    - You can only install one version of a package, so only one of these can be installed: doctrine/dbal[2.2.0-beta1, ..., 2.13.9, 3.0.0, ..., 3.5.x-dev].
    - pimcore/pimcore 11.x-dev requires doctrine/dbal ^3.4.2 -> satisfiable by doctrine/dbal[3.4.2, 3.4.3, 3.4.x-dev, 3.5.x-dev].
    - Root composer.json requires pimcore/pimcore 11.x-dev as 10.99.99 -> satisfiable by pimcore/pimcore[11.x-dev].

```

**Comment by dvesh3** [comment]
Fixed by #327
